doc/arch/adr-002-rest-api.md

```diff
@@ -0,0 +1,74 @@
+# Boundary Sync REST API Design
+
+## Context
+
+Based on the [data model](./adr-001-data-models.md), the following REST API design is proposed to access it:
+
+### Authentication
+
+| Path            | Method | Body      | Authentication | Responses                                             | Notes                                        |
+| --------------- | ------ | --------- | -------------- | ----------------------------------------------------- | -------------------------------------------- |
+| `/auth/login/`  | GET    | -         | Allow Any      | 200 OK,<br />401 Not Authorized                       | Returns loging JSON if current session valid |
+| `/auth/login/`  | POST   | `{ ... }` | Allow Any      | 200 OK,<br />400 Bad Request,<br />401 Not Authorized | Used for starting new valid session          |
+| `/auth/logout/` | POST   | -         | Logged In User | 200 OK                                                | Used for logging out                         |
+
+### Boundaries
```

Comment:
I'm a little conflicted about the Boundary / Submission separation. In our data model, the Boundary is the top-level entity, which may have one or more Submissions through its life. But in the UI, we're calling everything Submissions, so this is getting a little confusing.

Reply:
I'm willing to get rid of Boundary and just treat Utilities as boundaries (== groups of submissions).

Reply:
To flesh out that thought a little: otherwise, if we leave Boundary as a model, then I'm okay with the design in this ADR as is.

Reply:
A utility can have multiple boundaries submitted over the years, and each boundary is a group of submissions, so we can't make that direct link. Need something in between.

Reply:
I'm a bit confused by this. 

With `/boundaries/{id}/submit/`, is the `id` the submission ID or the boundary ID? If not the submission ID, then where is that data? In the body of the request? I don't think we can determine the submission ID just from the boundary ID.

> But in the UI, we're calling everything Submissions, so this is getting a little confusing.

The users aren't going to see the API routes, so why not use naming that reflects the data model?

Reply:
Just saw this comment: https://github.com/azavea/iow-boundary-tool/pull/115#discussion_r991251211

Is that the case with submissions too? We make them transparent to the front-end because the correct one can be determined in the back-end?

Reply:
> We make them transparent to the front-end because the correct one can be determined in the back-end?

Correct.

I'm imagining that POSTing to `/boundaries/` creates a Boundary and a Submission, and returns details for both. POSTING to `/boundaries/{id}/submit/` updates the Submission with `notes` and sets its `submitted_at` and `submitted_by`.

This reveals that we need an additional endpoint for creating a second draft submission. I'll add one.

Reply:
But only one draft can be in review at a time? (Because 0-1 reviews per boundary.)

Reply:
Correct, only one active submission (in draft or submitted state) at a time.

Reply:
Update from scrum: not merely one draft in review at a time, but one draft at a time _period_. Coworkers will be collaborating on the same draft.